Content.Server/Objectives/Systems/StealConditionSystem.cs

```diff
@@ -101,15 +105,19 @@ private float GetProgress(MindComponent mind, StealConditionComponent condition)
         //check stealAreas
         if (condition.CheckStealAreas)
         {
-            var areasQuery = AllEntityQuery<StealAreaComponent>();
-            while (areasQuery.MoveNext(out var uid, out var area))
+            var areasQuery = AllEntityQuery<TransformComponent, StealAreaComponent>();
+            while (areasQuery.MoveNext(out var uid, out var xform, out var area))
```

Comment:
Remember that this iterates literally every entity ingame.

Reply:
isnt the most common comp supposed to be first arg

Reply:
Without Arch it just gets the dictionary of the first component and then trycomps every single entity in that list.

Reply:
bruh